compiler/pipes/calc-real-defines-values.cpp

```diff
@@ -131,6 +131,9 @@ void CalcRealDefinesAndAssignModulitesF::process_define(DefinePtr def) {
     def->type() = DefineData::def_const;
     def->val = make_const.make_const(def->val);
     def->val->const_type = cnst_const_val;
+    if (auto s = def->val.try_as<op_define_val>()) {
```

Comment:
Where should I look?
Yes, it is necessary in my opinion. It is like a base case for calc-const-types pass

Reply:
It seems very strange, because for
```
const X = SOME_DEF;
```
then def->val is op_define_val, and const_type will be set, but for
```
const X = [SOME_DEF];
```
then nested op_define_val const_type won't be set.
Am I wrong?
Maybe, setitng const_type should be done in MakeConst manupilation?

Reply:
You are wrong because we have recursive `process_define_recursive()`.
I did by analogy with line 133. MakeConst may be used in other contexts and that is why I did so.